sdk-core/src/test/java/care/data4life/sdk/auth/UserServiceTest.kt

```diff
@@ -0,0 +1,251 @@
+        //then
+        testSubscriber?.assertValue(true)
+        verify(exactly = 1) { Log.error(error, "Version not supported") }
+    }
+
+    @Test
+    fun `Given VersionList is fetched, when version supported return true`() {
+        val versions: VersionList = mockk()
+        val response: Single<VersionList> = Single.just(versions)
+        val currentVersion = "1.9.0"
+        every { apiService.fetchVersionInfo() } returns response
+        every { versions.isSupported(currentVersion) } returns true
+        //when
+        val testSubscriber = userService.getVersionInfo("1.9.0")
+            ?.test()
+        //then
+        testSubscriber?.assertValue(true)
+            ?.assertNoErrors()
+
+    }
+
+    @Test
+    fun `logout should Return Success`() {
+        //given
+        every { apiService.logout(USER_ALIAS) } returns Completable.complete()
+        every { storage.clear() } just Runs
+        // when
+        val testSubscriber = userService.logout()
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertComplete()
+        verify { storage.clear() }
+    }
+
+    @Test
+    fun `logout should Return Exception`() {
+        //given
+        every { apiService.logout(USER_ALIAS) } returns Completable.error(Exception())
+        // when
+        val testSubscriber = userService.logout()
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertError(Exception::class.java)
+            .assertNotComplete()
+        verify { storage wasNot Called }
+    }
+
+    @Test
+    fun `finish Login should Return Boolean Success`() {
+        //given
+        val userInfo = mockk<UserInfo>()
+        every { userInfo.commonKey } returns mockk<EncryptedKey>()
+        every { userInfo.commonKeyId } returns "mockedCommonKeyId"
+        every { userInfo.tagEncryptionKey } returns mockk<EncryptedKey>()
+        every { userInfo.uid } returns "mockedUid"
+        every { apiService.fetchUserInfo(USER_ALIAS) } returns Single.just(userInfo)
+        every { storage.storeSecret("userAlias_user_id", "mockedUid") } just Runs
+        every { cryptoService.storeCommonKey("mockedCommonKeyId", any()) } just Runs
+        every { cryptoService.storeCurrentCommonKeyId("mockedCommonKeyId") } just Runs
+        every { cryptoService.storeTagEncryptionKey(any()) } just Runs
+        every { cryptoService.fetchGCKeyPair() } returns Single.just(mockk<GCKeyPair>())
+        every {
+            cryptoService.asymDecryptSymetricKey(any(), any())
+        } returns Single.just(mockk<GCKey>())
+        every {
+            cryptoService.symDecryptSymmetricKey(any(), any())
+        } returns Single.just(mockk<GCKey>())
+        // when
+        val testSubscriber = userService.finishLogin(true)
+            .test()
+
+
+        // then
+        testSubscriber
+            .assertValue { aBoolean: Boolean? -> aBoolean!! }
```

Comment:
You can replace `aBoolean: Boolean? -> aBoolean!!` simply by `it` or write `aBoolean -> aBoolean` to get rid of the unnecessary typing.